cdap-data-fabric/src/main/java/co/cask/cdap/data/stream/StreamUtils.java

```diff
@@ -47,6 +49,10 @@
  */
 public final class StreamUtils {
 
+  // The directory name for storing stream files that are pending for deletion
+  // StreamId cannot have "." there, so it's safe that it won't clash with any stream name
+  private static final String DELETED = ".deleted";
```

Comment:
CDAP is also for people who are not familiar with Hadoop or don't want/need to deal with Hadoop :)